orderer/consensus/smartbft/util.go

```diff
@@ -283,6 +283,25 @@ func (ri *RequestInspector) RequestID(rawReq []byte) types.RequestInfo {
 	if err != nil {
 		return types.RequestInfo{}
 	}
+
+	if req.chHdr.Type == int32(cb.HeaderType_CONFIG) {
+		configEnvelope := &cb.ConfigEnvelope{}
```

Comment:
why do such a deep unmarshal if we only care about the top most signature at the envelope level? 

RequestInfo is a tuple of client identifier (can remain the same) and a request ID.

The request ID for config updates can be simply a hash over the data field in the payload. 
This way, we ignore both the header and the signature fields, which are what can change when the orderer re-evaluates the request.

Reply:
In your case the client identifier of all transactions will be different - the order identifier

So you don't have to redo a lot of things.
I learned that in every transaction that changes the channel configuration, there is the same, original request to change the channel in an unchanged form. 
It is located in the LastUpdate field. 
I use LastUpdate to make RequestInfo.

Reply:
The orderer identifier is in the header, that's why I suggested to ignore the header and only look at the data in the payload. 

The channel config stuff is all in the data field in the payload. 
Why do you need to dig so deep in the data? Just take the data as-is.

Reply:
I'm sorry, I guess I don't quite get it.

RequestInfo is a tuple of client id and request id.

It should be the same for channel update transactions.

Assuming the request id I will do as you say.
But what do you suggest to put in the client id field?

Reply:
If client id is left as it is, it will be different for all transactions. Therefore, from the point of view of the consensus library, they will be different transactions. 
I just wanted these transactions to have the same id from the consensus point of view

Reply:
Just put "" in it, for simplicity.

Reply:
ок

Reply:
@yacovm Alas, that option failed. Because `payload.Data` on each order is different. The data inside are the same, but after conversion to []byte they are different. Most likely it is related to the data size and conversion to protobuf and back.

Here is an example of a test that shows it:
```
func TestProtobuf(t *testing.T) {
	var TxIDStrEtl string

	for i := 0; i < 2; i++ {
		s := "{ \"channel_group\": { \"groups\": { \"Application\": { \"groups\": { \"Org1\": { \"groups\": {}, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Endorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 }, { \"signed_by\": 2 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 } ] } }, \"version\": 0 } }, \"version\": \"0\" } }, \"values\": { \"AnchorPeers\": { \"mod_policy\": \"Admins\", \"value\": { \"anchor_peers\": [ { \"host\": \"127.0.0.1\", \"port\": 29020 } ] }, \"version\": \"0\" }, \"MSP\": { \"mod_policy\": \"Admins\", \"value\": { \"config\": { \"admins\": [], \"crypto_config\": { \"identity_identifier_hash_function\": \"SHA256\", \"signature_hash_family\": \"SHA2\" }, \"fabric_node_ous\": { \"admin_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"admin\" }, \"client_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"client\" }, \"enable\": true, \"orderer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"orderer\" }, \"peer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"peer\" } }, \"intermediate_certs\": [], \"name\": \"Org1MSP\", \"organizational_unit_identifiers\": [], \"revocation_list\": [], \"root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\" ], \"signing_identity\": null, \"tls_intermediate_certs\": [], \"tls_root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNWekNDQWYyZ0F3SUJBZ0lRTUJGYmVTZTQzNzFGR0dTYm0rSE1mekFLQmdncWhrak9QUVFEQWpCMk1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTVM1bGVHRnRjR3hsTG1OdmJURWZNQjBHQTFVRUF4TVdkR3h6ClkyRXViM0puTVM1bGVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXoKTURCYU1IWXhDekFKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSApFdzFUWVc0Z1JuSmhibU5wYzJOdk1Sa3dGd1lEVlFRS0V4QnZjbWN4TG1WNFlXMXdiR1V1WTI5dE1SOHdIUVlEClZRUURFeFowYkhOallTNXZjbWN4TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMEQKQVFjRFFnQUVmaVhrL1JMRGhJWjRTeFM3NU5hYktkRSszUTBkWVJlU3kySklqQ0YwUk5IRGE5K3Zxa0ZKTHluMQpzVmlzdWgwR0RmQlNjR1NEL056UFZRTVVJZEtqazZOdE1Hc3dEZ1lEVlIwUEFRSC9CQVFEQWdHbU1CMEdBMVVkCkpRUVdNQlFHQ0NzR0FRVUZCd01DQmdnckJnRUZCUWNEQVRBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUNrR0ExVWQKRGdRaUJDQlhwWHptSXE5L05KNUsvT1l6SnV1YVZ4cXlqdFRUWXpXbERzeGVKRXY5Y3pBS0JnZ3Foa2pPUFFRRApBZ05JQURCRkFpQXFYUSs4cHVSZzRqTFUwVmZid2p4clRjMU9xbmEwdHUzYnBtSTRHQVlyUkFJaEFLQ2RiVUI2ClJrVVV4UzVNakpNbmV5bWtLQ3oxNmI0KzZKMzN1ekVSVGJHeQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" ] }, \"type\": 0 }, \"version\": \"0\" } }, \"version\": \"0\" }, \"Org2\": { \"groups\": {}, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Endorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 }, { \"signed_by\": 2 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 } ] } }, \"version\": 0 } }, \"version\": \"0\" } }, \"values\": { \"AnchorPeers\": { \"mod_policy\": \"Admins\", \"value\": { \"anchor_peers\": [ { \"host\": \"127.0.0.1\", \"port\": 29025 } ] }, \"version\": \"0\" }, \"MSP\": { \"mod_policy\": \"Admins\", \"value\": { \"config\": { \"admins\": [], \"crypto_config\": { \"identity_identifier_hash_function\": \"SHA256\", \"signature_hash_family\": \"SHA2\" }, \"fabric_node_ous\": { \"admin_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"admin\" }, \"client_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"client\" }, \"enable\": true, \"orderer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"orderer\" }, \"peer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"peer\" } }, \"intermediate_certs\": [], \"name\": \"Org2MSP\", \"organizational_unit_identifiers\": [], \"revocation_list\": [], \"root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\" ], \"signing_identity\": null, \"tls_intermediate_certs\": [], \"tls_root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNWekNDQWY2Z0F3SUJBZ0lSQU8rVnlBd2YwUWx5MTdaR2xkWk9kU2d3Q2dZSUtvWkl6ajBFQXdJd2RqRUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpJdVpYaGhiWEJzWlM1amIyMHhIekFkQmdOVkJBTVRGblJzCmMyTmhMbTl5WnpJdVpYaGhiWEJzWlM1amIyMHdIaGNOTWpNeE1qSTNNVFkwTXpBd1doY05Nek14TWpJME1UWTAKTXpBd1dqQjJNUXN3Q1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRQpCeE1OVTJGdUlFWnlZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWZNQjBHCkExVUVBeE1XZEd4elkyRXViM0puTWk1bGVHRnRjR3hsTG1OdmJUQlpNQk1HQnlxR1NNNDlBZ0VHQ0NxR1NNNDkKQXdFSEEwSUFCTi96a2NwRHNwNmM5eTlTOGsxWityeXE4aE5vTjZObjFpOFFtcURMQjJXR2ptWThWam1nSythKwpKWmRqclA2dG1hWlVjYldjcWUxOTIyQm9ZcWhWcGdpamJUQnJNQTRHQTFVZER3RUIvd1FFQXdJQnBqQWRCZ05WCkhTVUVGakFVQmdnckJnRUZCUWNEQWdZSUt3WUJCUVVIQXdFd0R3WURWUjBUQVFIL0JBVXdBd0VCL3pBcEJnTlYKSFE0RUlnUWd1WEI4dDBvTzRnMDdTSHhRMk0rTi9ZZXhTdFJEMFUwSmZHaE0rQmZFdUgwd0NnWUlLb1pJemowRQpBd0lEUndBd1JBSWdDMkVVZW14dDBoaWdQdWZtN0VQVjFWc3U0NU5JSlNRckhDeC9pcmlTK2FFQ0lFbEwvNTRjClhzVnhobVhhVXRmckdrZm5sK01JbllURmpjdjJ3STJLZUdpRAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" ] }, \"type\": 0 }, \"version\": \"0\" } }, \"version\": \"0\" } }, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Admins\" } }, \"version\": \"0\" }, \"Endorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Endorsement\" } }, \"version\": \"0\" }, \"LifecycleEndorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Endorsement\" } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Readers\" } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Writers\" } }, \"version\": \"0\" } }, \"values\": { \"Capabilities\": { \"mod_policy\": \"Admins\", \"value\": { \"capabilities\": { \"V2_0\": {} } }, \"version\": \"0\" } }, \"version\": \"0\" }, \"Orderer\": { \"groups\": { \"OrdererOrg\": { \"groups\": {}, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"OrdererMSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"OrdererMSP\", \"role\": \"MEMBER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"OrdererMSP\", \"role\": \"MEMBER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" } }, \"values\": { \"Endpoints\": { \"mod_policy\": \"Admins\", \"value\": { \"addresses\": [ \"127.0.0.1:29000\", \"127.0.0.1:29005\", \"127.0.0.1:29010\", \"127.0.0.1:29015\" ] }, \"version\": \"0\" }, \"MSP\": { \"mod_policy\": \"Admins\", \"value\": { \"config\": { \"admins\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNDekNDQWJHZ0F3SUJBZ0lSQUpaUGFLTk9TWG01T2g0bjJLWXNjWjh3Q2dZSUtvWkl6ajBFQXdJd2FURUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJjd0ZRWURWUVFERXc1allTNWxlR0Z0CmNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmFNRll4Q3pBSkJnTlYKQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcApjMk52TVJvd0dBWURWUVFEREJGQlpHMXBia0JsZUdGdGNHeGxMbU52YlRCWk1CTUdCeXFHU000OUFnRUdDQ3FHClNNNDlBd0VIQTBJQUJGWnFPdzJEcWpjaEFPOHpkM0YwNFhBR01xQ0FEeHBRdWZBeFErcGE0Smt3VDVya2VZK2YKNWxwZEZKOWJqWjZEWmEzdlRZNk1PNzhqamI0K2ZEZExrL1dqVFRCTE1BNEdBMVVkRHdFQi93UUVBd0lIZ0RBTQpCZ05WSFJNQkFmOEVBakFBTUNzR0ExVWRJd1FrTUNLQUlLT0V4RURoSmRJLy9rQU9oYnQrcFdrcDQ5VUJuWlRBCmdxSHFESjdqUEdaL01Bb0dDQ3FHU000OUJBTUNBMGdBTUVVQ0lRQ1BxVnRQWndEdHY1TkoxelIvOExzNjJLMjEKSHMxRnhrS3NDTDhqWmJSaUVnSWdPc3JvQ3huNWhXTDVGTlI2Y1NpWEc5bHVYYVE4QkJFeXM5bVJFSk0veGRZPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" ], \"crypto_config\": { \"identity_identifier_hash_function\": \"SHA256\", \"signature_hash_family\": \"SHA2\" }, \"fabric_node_ous\": null, \"intermediate_certs\": [], \"name\": \"OrdererMSP\", \"organizational_unit_identifiers\": [], \"revocation_list\": [], \"root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNQakNDQWVPZ0F3SUJBZ0lRWEpmMjhlcGZiQ0ZUZzlzZXZyL2JRVEFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dhVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4RkRBU0JnTlZCQW9UQzJWNFlXMXdiR1V1WTI5dE1SY3dGUVlEVlFRREV3NWpZUzVsZUdGdGNHeGxMbU52CmJUQlpNQk1HQnlxR1NNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJNSS90NjdhdUlPT1lrOCtyN3p2UkhiRDQ4S0sKWGNtYUdPVmQwbGpyMlY0Q21uQjdCNk4wWTd1d001N0U0QlBMVVFuRmFPZkhvWG1kVThvNGZ1b21GZGFqYlRCcgpNQTRHQTFVZER3RUIvd1FFQXdJQnBqQWRCZ05WSFNVRUZqQVVCZ2dyQmdFRkJRY0RBZ1lJS3dZQkJRVUhBd0V3CkR3WURWUjBUQVFIL0JBVXdBd0VCL3pBcEJnTlZIUTRFSWdRZ280VEVRT0VsMGovK1FBNkZ1MzZsYVNuajFRR2QKbE1DQ29lb01udU04Wm44d0NnWUlLb1pJemowRUF3SURTUUF3UmdJaEFKSkJCWUI5KzVKWmZvdGR1TkxXcTBPWAo4WTlEYml0QUQxa0ZXL2xVaHUxbEFpRUFyQWc3WXpmVGdPL1o2Nlg3d0UwVmJ0cWQ1UXFzRzZQZFBFdlViY21ICmZCaz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\" ], \"signing_identity\": null, \"tls_intermediate_certs\": [], \"tls_root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNRakNDQWVtZ0F3SUJBZ0lRTnVQWjlaREJTcloxclMxd3dDS00vREFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93YkRFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEZEQVNCZ05WQkFvVEMyVjRZVzF3YkdVdVkyOXRNUm93R0FZRFZRUURFeEYwYkhOallTNWxlR0Z0CmNHeGxMbU52YlRCWk1CTUdCeXFHU000OUFnRUdDQ3FHU000OUF3RUhBMElBQk5nTEpBQ0NPb3MwdGo1Ri96UUoKak81a2IrWm5oKzVyK0NrL01SOXU0VFVnVm03MHdYZXJyQlFra0s2MzdFbXFlMnFvQWY1V1JqeFVULzdEcVB0ZgpnWG1qYlRCck1BNEdBMVVkRHdFQi93UUVBd0lCcGpBZEJnTlZIU1VFRmpBVUJnZ3JCZ0VGQlFjREFnWUlLd1lCCkJRVUhBd0V3RHdZRFZSMFRBUUgvQkFVd0F3RUIvekFwQmdOVkhRNEVJZ1FnallLSndVWjB5NnhXWFZQVnFVR1EKdUE5eWxuLzZHVXh5WDRIeUdVTURtMWt3Q2dZSUtvWkl6ajBFQXdJRFJ3QXdSQUlnUzFKdFdUenFNMG8zMHVqTQpYeTJXOTVnTzcvdUNnMVpVWG9ZWlo5OTdUemdDSUg0dU8xOWZXL2k2UTNSd09tdk1uRm5DVWo0dFNWVFpzSXhICk5tNzdHSGJPCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\" ] }, \"type\": 0 }, \"version\": \"0\" } }, \"version\": \"0\" } }, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Admins\" } }, \"version\": \"0\" }, \"BlockValidation\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRQ3lkRXdicW43Yy9QbU1kMFV1Sm1UakFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl4TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUV3b1U0ejRVMEZTUTFQaUI1Qm5XUDVEZ1BOQVlEc2QrUVhiQ3VPTndrM3RWNGkvaGgKZnh5aVFaS0RzUEUzUU5HV0JuaFFrSVpuZVF0QTQwVWNDTEdmMjZOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTjZ6ZCtJRXVEcUViMVJiakh6b2tKaUMKdUZRYm1wa2Y1aUhkSnc5Zm5kMDNBaUFkNEVZNmZvVTAzSG9vN2N1TjVsN2tubnZVVnFtZWlUZEVJVTE2N0g1cQpXZz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" }, { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEakNDQWJTZ0F3SUJBZ0lSQU4rZzVTNkZFMk1ldlNaZ1ZCT3Aybmd3Q2dZSUtvWkl6ajBFQXdJd2FURUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJjd0ZRWURWUVFERXc1allTNWxlR0Z0CmNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmFNRmt4Q3pBSkJnTlYKQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcApjMk52TVIwd0d3WURWUVFERXhSdmNtUmxjbVZ5TWk1bGVHRnRjR3hsTG1OdmJUQlpNQk1HQnlxR1NNNDlBZ0VHCkNDcUdTTTQ5QXdFSEEwSUFCQzFaNTlSbmNYdVI1OG9DVzErV1RWK1hsOGpxY2NCbzVkbDB5L0h6VDZrd3JDQzcKTytmK2x5bEFvZTgxTGdqdVIvNEtMcTBLQ0pVZnhnZkpGMDlXSGV5alRUQkxNQTRHQTFVZER3RUIvd1FFQXdJSApnREFNQmdOVkhSTUJBZjhFQWpBQU1Dc0dBMVVkSXdRa01DS0FJS09FeEVEaEpkSS8va0FPaGJ0K3BXa3A0OVVCCm5aVEFncUhxREo3alBHWi9NQW9HQ0NxR1NNNDlCQU1DQTBnQU1FVUNJUUNiWWxjaU9WaGpueUorQmhoYXZNQlcKKzlzUGRGYnh4MFpEWjdtVlNHT1Z6QUlnRzVkS2wzemV0bS9yRXBpS2ZjOXNTVTZ1OXgrMWJTTGtiR1RCQnpCYgpWYkU9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" }, { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNDekNDQWJPZ0F3SUJBZ0lRZTdObHVkYW9oeHhwbmdhM3padGlyekFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl6TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVlTHJpdnNOT0VoT2xTTyt3eFNGemlzU3R2VGRZa2taTDRnaG1VN0VmVVo0K0hsL3AKWU0wZ28xZHlDOFRyYW9sNDh6cnhidXAwSXk3TzNuSmpiOWdaQnFOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEUmdBd1F3SWZKREJwbURDWC9WSEFCNmd1andFT3FxdUkKZEtaRGxzWklaWHAwTkVLb3dBSWdhVWJMNlZ4dkx0Q2dYNjdCMDBFa1ZzOXNjZDMxTHFxYlZTcHBOZnR2bVd3PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" }, { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRRGJvNFB5VzlObnIwOSsvMUcyeHBFREFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEkwTG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVqaWlJdkU4cG5qZlNZeGZCdURDbExqOUdZNXRYb2k0MmpBcVdjRE5UV2JYYTI4NVYKb0VXd2FDenpKeXF1WUE5QzZXM2pRQTV1U1dtWjNBWVlKUFZGV3FOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTThWQWFCTHF2TU14bEI0SzlsSE44RjYKNFhwWlU5ZmxPS3pLUHowdFFEQ1pBaUFYSzJoM1FuaFAyb2VGT21HYzYreE1oVHhOVDFYemxxZWF2ZWlWamRpZgpqdz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" } ], \"rule\": { \"n_out_of\": { \"n\": 3, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 }, { \"signed_by\": 2 }, { \"signed_by\": 3 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Readers\" } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Writers\" } }, \"version\": \"0\" } }, \"values\": { \"BatchSize\": { \"mod_policy\": \"Admins\", \"value\": { \"absolute_max_bytes\": 1000000, \"max_message_count\": 1, \"preferred_max_bytes\": 524288 }, \"version\": \"1\" }, \"BatchTimeout\": { \"mod_policy\": \"Admins\", \"value\": { \"timeout\": \"1s\" }, \"version\": \"0\" }, \"Capabilities\": { \"mod_policy\": \"Admins\", \"value\": { \"capabilities\": { \"V2_0\": {} } }, \"version\": \"0\" }, \"ChannelRestrictions\": { \"mod_policy\": \"Admins\", \"value\": null, \"version\": \"0\" }, \"ConsensusType\": { \"mod_policy\": \"Admins\", \"value\": { \"metadata\": { \"collect_timeout\": \"1s\", \"decisions_per_leader\": \"0\", \"incoming_message_buffer_size\": \"200\", \"leader_heartbeat_count\": \"10\", \"leader_heartbeat_timeout\": \"1m0s\", \"leader_rotation\": \"ROTATION_UNSPECIFIED\", \"request_auto_remove_timeout\": \"3m0s\", \"request_batch_max_bytes\": \"10485760\", \"request_batch_max_count\": \"100\", \"request_batch_max_interval\": \"50ms\", \"request_complain_timeout\": \"20s\", \"request_forward_timeout\": \"2s\", \"request_max_bytes\": \"0\", \"request_pool_size\": \"400\", \"speed_up_view_change\": false, \"sync_on_start\": false, \"view_change_resend_interval\": \"5s\", \"view_change_timeout\": \"20s\" }, \"state\": \"STATE_NORMAL\", \"type\": \"BFT\" }, \"version\": \"0\" }, \"Orderers\": { \"mod_policy\": \"Admins\", \"value\": { \"consenter_mapping\": [ { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlhZ0F3SUJBZ0lSQU5rMFcrNTBwdUJkSmFFK0VmK284c3d3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQktiSXFLZTVZbG5vY3lCV3lJczZpaHJpQUY3UllqSjZXN2lDS094b3YyV1EKbHNHazA1R2c5N21lcnBsOEZpSEJGSjJ3cWs5cDhXSG9HbWVkUFVoNlQrU2pnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJeExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlNWUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwY0FNRVFDCklCandvaitiV1ZkeTZQMUlBMGw3a2FpSEYwOGhyQTRCRTY0SktLZzZyblhQQWlCOUg1NEVwZTI0L2JTWkwvNC8KVjdnY2pIQ3lhcXVwUEp6R3M2VFdGSnhYSUE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"host\": \"127.0.0.1\", \"id\": 1, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRQ3lkRXdicW43Yy9QbU1kMFV1Sm1UakFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl4TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUV3b1U0ejRVMEZTUTFQaUI1Qm5XUDVEZ1BOQVlEc2QrUVhiQ3VPTndrM3RWNGkvaGgKZnh5aVFaS0RzUEUzUU5HV0JuaFFrSVpuZVF0QTQwVWNDTEdmMjZOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTjZ6ZCtJRXVEcUViMVJiakh6b2tKaUMKdUZRYm1wa2Y1aUhkSnc5Zm5kMDNBaUFkNEVZNmZvVTAzSG9vN2N1TjVsN2tubnZVVnFtZWlUZEVJVTE2N0g1cQpXZz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"msp_id\": \"OrdererMSP\", \"port\": 29003, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlhZ0F3SUJBZ0lSQU5rMFcrNTBwdUJkSmFFK0VmK284c3d3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQktiSXFLZTVZbG5vY3lCV3lJczZpaHJpQUY3UllqSjZXN2lDS094b3YyV1EKbHNHazA1R2c5N21lcnBsOEZpSEJGSjJ3cWs5cDhXSG9HbWVkUFVoNlQrU2pnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJeExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlNWUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwY0FNRVFDCklCandvaitiV1ZkeTZQMUlBMGw3a2FpSEYwOGhyQTRCRTY0SktLZzZyblhQQWlCOUg1NEVwZTI0L2JTWkwvNC8KVjdnY2pIQ3lhcXVwUEp6R3M2VFdGSnhYSUE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" }, { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlXZ0F3SUJBZ0lRZUlVcGZPd3RGR0tLQ0Q3MFRKN1NOekFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFRjNkNFhzcG5FVk94Y2lnekJmMEg1TWFxY1JsMHBLYWV0alZXdHpIZzRGZGYKYkcyQXFTdE1VcG0xYmNkb3VEZFNESUd0ZUFIckJxcFAwNWZXcXhKK0NhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pJdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXlnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFNBQXdSUUloCkFNK2szcE9xa2YrNUlvcGplSW5FaDY0djRwd0t5UWc4cG1FU3E2cmNGV1BJQWlBaXUrMEZnSTNlcDNRTDg1emwKdmsySjdxRmhsUHdyL28rYzQxUzQxaVpuSkE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"host\": \"127.0.0.1\", \"id\": 2, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEakNDQWJTZ0F3SUJBZ0lSQU4rZzVTNkZFMk1ldlNaZ1ZCT3Aybmd3Q2dZSUtvWkl6ajBFQXdJd2FURUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJjd0ZRWURWUVFERXc1allTNWxlR0Z0CmNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmFNRmt4Q3pBSkJnTlYKQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcApjMk52TVIwd0d3WURWUVFERXhSdmNtUmxjbVZ5TWk1bGVHRnRjR3hsTG1OdmJUQlpNQk1HQnlxR1NNNDlBZ0VHCkNDcUdTTTQ5QXdFSEEwSUFCQzFaNTlSbmNYdVI1OG9DVzErV1RWK1hsOGpxY2NCbzVkbDB5L0h6VDZrd3JDQzcKTytmK2x5bEFvZTgxTGdqdVIvNEtMcTBLQ0pVZnhnZkpGMDlXSGV5alRUQkxNQTRHQTFVZER3RUIvd1FFQXdJSApnREFNQmdOVkhSTUJBZjhFQWpBQU1Dc0dBMVVkSXdRa01DS0FJS09FeEVEaEpkSS8va0FPaGJ0K3BXa3A0OVVCCm5aVEFncUhxREo3alBHWi9NQW9HQ0NxR1NNNDlCQU1DQTBnQU1FVUNJUUNiWWxjaU9WaGpueUorQmhoYXZNQlcKKzlzUGRGYnh4MFpEWjdtVlNHT1Z6QUlnRzVkS2wzemV0bS9yRXBpS2ZjOXNTVTZ1OXgrMWJTTGtiR1RCQnpCYgpWYkU9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"msp_id\": \"OrdererMSP\", \"port\": 29008, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlXZ0F3SUJBZ0lRZUlVcGZPd3RGR0tLQ0Q3MFRKN1NOekFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFRjNkNFhzcG5FVk94Y2lnekJmMEg1TWFxY1JsMHBLYWV0alZXdHpIZzRGZGYKYkcyQXFTdE1VcG0xYmNkb3VEZFNESUd0ZUFIckJxcFAwNWZXcXhKK0NhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pJdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXlnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFNBQXdSUUloCkFNK2szcE9xa2YrNUlvcGplSW5FaDY0djRwd0t5UWc4cG1FU3E2cmNGV1BJQWlBaXUrMEZnSTNlcDNRTDg1emwKdmsySjdxRmhsUHdyL28rYzQxUzQxaVpuSkE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" }, { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmakNDQWlXZ0F3SUJBZ0lRVGRJa1Q1TU5YVW8yNXI4NUJNbHVQakFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJekxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFeEowOTE1L2N1Rk5RczdLeUhCV3JyWTVHb1hvWkxMc1VjRHJMTFBhd21odmwKemllbG9nYVE1Tll1WmFYVk5IRm5hcFAvTy94K09Ncmlvejh2ZDNhWkhhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pNdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXpnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFJ3QXdSQUlnCmVNTnNVS29lMmdXQnpVdzRIeWpkN3FqeUNlUmwxaUtwejQxcmtGSnBNRWNDSUhtdXp1Yng1ZUljdGd5ZHNaUHMKR0xvOERlSzNPMEVWdHJESkNrQ2YzL2RrCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"host\": \"127.0.0.1\", \"id\": 3, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNDekNDQWJPZ0F3SUJBZ0lRZTdObHVkYW9oeHhwbmdhM3padGlyekFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl6TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVlTHJpdnNOT0VoT2xTTyt3eFNGemlzU3R2VGRZa2taTDRnaG1VN0VmVVo0K0hsL3AKWU0wZ28xZHlDOFRyYW9sNDh6cnhidXAwSXk3TzNuSmpiOWdaQnFOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEUmdBd1F3SWZKREJwbURDWC9WSEFCNmd1andFT3FxdUkKZEtaRGxzWklaWHAwTkVLb3dBSWdhVWJMNlZ4dkx0Q2dYNjdCMDBFa1ZzOXNjZDMxTHFxYlZTcHBOZnR2bVd3PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"msp_id\": \"OrdererMSP\", \"port\": 29013, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmakNDQWlXZ0F3SUJBZ0lRVGRJa1Q1TU5YVW8yNXI4NUJNbHVQakFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJekxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFeEowOTE1L2N1Rk5RczdLeUhCV3JyWTVHb1hvWkxMc1VjRHJMTFBhd21odmwKemllbG9nYVE1Tll1WmFYVk5IRm5hcFAvTy94K09Ncmlvejh2ZDNhWkhhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pNdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXpnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFJ3QXdSQUlnCmVNTnNVS29lMmdXQnpVdzRIeWpkN3FqeUNlUmwxaUtwejQxcmtGSnBNRWNDSUhtdXp1Yng1ZUljdGd5ZHNaUHMKR0xvOERlSzNPMEVWdHJESkNrQ2YzL2RrCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\" }, { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNnRENDQWlhZ0F3SUJBZ0lSQU8vRjJYbjBPMi9CaDVxYlRwQWdkVTB3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU5DNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQkpKeDdCZXlpK3Rtbk9zM3pSMTVlWURYa0M0WEU0MHJYeWg4Ylpsa1pDeXIKbG4wcVBZZDMybzZ1WlRDVThyRFZsejhyaTBzT2RLUVNvNTcvUkRneVZ5ZWpnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJMExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlOSUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwZ0FNRVVDCklRREJ0ZjY1eGdUYTJuREFVZjZiN1JCWlVKZ29GYlNzTkVINFFvaWJydDlWNHdJZ0g0cWJRQ2tWcEo0UkFKZm8KZUNMNEZ0Qk5ySE9CV0xqamt6NGxIYmdmVWYwPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"host\": \"127.0.0.1\", \"id\": 4, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRRGJvNFB5VzlObnIwOSsvMUcyeHBFREFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEkwTG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVqaWlJdkU4cG5qZlNZeGZCdURDbExqOUdZNXRYb2k0MmpBcVdjRE5UV2JYYTI4NVYKb0VXd2FDenpKeXF1WUE5QzZXM2pRQTV1U1dtWjNBWVlKUFZGV3FOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTThWQWFCTHF2TU14bEI0SzlsSE44RjYKNFhwWlU5ZmxPS3pLUHowdFFEQ1pBaUFYSzJoM1FuaFAyb2VGT21HYzYreE1oVHhOVDFYemxxZWF2ZWlWamRpZgpqdz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"msp_id\": \"OrdererMSP\", \"port\": 29018, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNnRENDQWlhZ0F3SUJBZ0lSQU8vRjJYbjBPMi9CaDVxYlRwQWdkVTB3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU5DNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQkpKeDdCZXlpK3Rtbk9zM3pSMTVlWURYa0M0WEU0MHJYeWg4Ylpsa1pDeXIKbG4wcVBZZDMybzZ1WlRDVThyRFZsejhyaTBzT2RLUVNvNTcvUkRneVZ5ZWpnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJMExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlOSUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwZ0FNRVVDCklRREJ0ZjY1eGdUYTJuREFVZjZiN1JCWlVKZ29GYlNzTkVINFFvaWJydDlWNHdJZ0g0cWJRQ2tWcEo0UkFKZm8KZUNMNEZ0Qk5ySE9CV0xqamt6NGxIYmdmVWYwPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" } ] }, \"version\": \"0\" } }, \"version\": \"0\" } }, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Admins\" } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Readers\" } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Writers\" } }, \"version\": \"0\" } }, \"values\": { \"BlockDataHashingStructure\": { \"mod_policy\": \"Admins\", \"value\": { \"width\": 4294967295 }, \"version\": \"0\" }, \"Capabilities\": { \"mod_policy\": \"Admins\", \"value\": { \"capabilities\": { \"V3_0\": {} } }, \"version\": \"0\" }, \"HashingAlgorithm\": { \"mod_policy\": \"Admins\", \"value\": { \"name\": \"SHA256\" }, \"version\": \"0\" }, \"OrdererAddresses\": { \"mod_policy\": \"/Channel/Orderer/Admins\", \"value\": { \"addresses\": [ \"127.0.0.1:29000\", \"127.0.0.1:29005\", \"127.0.0.1:29010\", \"127.0.0.1:29015\" ] }, \"version\": \"0\" } }, \"version\": \"0\" }, \"sequence\": \"1\" } "
		b := bytes.NewBufferString(s)

		c := &cb.Config{}
		err := protolator.DeepUnmarshalJSON(b, c)
		assert.NoError(t, err)

		m, err := proto.Marshal(c)
		assert.NoError(t, err)

		txID := sha256.Sum256(m)
		TxIDStr := hex.EncodeToString(txID[:])

		if TxIDStrEtl == "" {
			TxIDStrEtl = TxIDStr
			continue
		}

		assert.Equal(t, TxIDStrEtl, TxIDStr)
	}
}

```

Reply:
Alas, you haven't made the code change according to what I suggested.

Look at [this line](https://github.com/hyperledger/fabric/blob/main/orderer/consensus/smartbft/configverifier.go#L206) in [verifyConfigUpdateMsg](https://github.com/hyperledger/fabric/blob/main/orderer/consensus/smartbft/configverifier.go#L171C34-L171C55). 

If the processing of the config in orderers was not deterministic as you alleged, then every config update sent by an orderer would have resulted in a view change.

Reply:
Are you implying that the config processing uses protolator? That's highly unlikely considering the fact that protolator packages import the configtx so even though I haven't looked, I'm pretty sure that configtx doesn't import protolator otherwise there would be a dependency cycle.

Reply:
No. I made your changes first, took the hash off the payload.Data. Then I ran the test (without protolator) and each order returned a different hash. 

I started searching. And I found it - it's about `m, err := proto.Marshal(c)`. On each orderer it returned a different slice.

Reply:
I redid the example.
c1 and c2 are equal.
But m1 and m2 are no equal.

```
func TestProtobuf(t *testing.T) {
	s := "{ \"channel_group\": { \"groups\": { \"Application\": { \"groups\": { \"Org1\": { \"groups\": {}, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Endorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 }, { \"signed_by\": 2 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org1MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 } ] } }, \"version\": 0 } }, \"version\": \"0\" } }, \"values\": { \"AnchorPeers\": { \"mod_policy\": \"Admins\", \"value\": { \"anchor_peers\": [ { \"host\": \"127.0.0.1\", \"port\": 29020 } ] }, \"version\": \"0\" }, \"MSP\": { \"mod_policy\": \"Admins\", \"value\": { \"config\": { \"admins\": [], \"crypto_config\": { \"identity_identifier_hash_function\": \"SHA256\", \"signature_hash_family\": \"SHA2\" }, \"fabric_node_ous\": { \"admin_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"admin\" }, \"client_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"client\" }, \"enable\": true, \"orderer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"orderer\" }, \"peer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"peer\" } }, \"intermediate_certs\": [], \"name\": \"Org1MSP\", \"organizational_unit_identifiers\": [], \"revocation_list\": [], \"root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZpZ0F3SUJBZ0lSQUlyQnEwZ3lWaXlnYzV0MzMwdzN2WE13Q2dZSUtvWkl6ajBFQXdJd2N6RUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpFdVpYaGhiWEJzWlM1amIyMHhIREFhQmdOVkJBTVRFMk5oCkxtOXlaekV1WlhoaGJYQnNaUzVqYjIwd0hoY05Nak14TWpJM01UWTBNekF3V2hjTk16TXhNakkwTVRZME16QXcKV2pCek1Rc3dDUVlEVlFRR0V3SlZVekVUTUJFR0ExVUVDQk1LUTJGc2FXWnZjbTVwWVRFV01CUUdBMVVFQnhNTgpVMkZ1SUVaeVlXNWphWE5qYnpFWk1CY0dBMVVFQ2hNUWIzSm5NUzVsZUdGdGNHeGxMbU52YlRFY01Cb0dBMVVFCkF4TVRZMkV1YjNKbk1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5QWdFR0NDcUdTTTQ5QXdFSEEwSUEKQkZZZHcvbWZlVW9HMUZKZUlvVnljS2diQlFJbzFYOG9iaTMvR0dzOElMRE1CS3RiUmtPZ0xBTHp5OXVhOTNLRwowYm91d2FaRGtENFdybk0zM2dHT0NGU2piVEJyTUE0R0ExVWREd0VCL3dRRUF3SUJwakFkQmdOVkhTVUVGakFVCkJnZ3JCZ0VGQlFjREFnWUlLd1lCQlFVSEF3RXdEd1lEVlIwVEFRSC9CQVV3QXdFQi96QXBCZ05WSFE0RUlnUWcKQmRMZzlJLzEremIybXZtVW9xNVlrZlh3NlN0a3JvZFZBZE1Xd2VjcW05Z3dDZ1lJS29aSXpqMEVBd0lEUndBdwpSQUlnS1ZLd0hsQWc1Z0dqN0V5VjgvenZSN0E0cms1bENtN3VHMWlHWmRKVkNDMENJQUVUWWkvMEthWjFKM0J4CnhhRThZbk1lTVlpWVAwaWVUcytvUXdOVUtIVGgKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\" ], \"signing_identity\": null, \"tls_intermediate_certs\": [], \"tls_root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNWekNDQWYyZ0F3SUJBZ0lRTUJGYmVTZTQzNzFGR0dTYm0rSE1mekFLQmdncWhrak9QUVFEQWpCMk1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTVM1bGVHRnRjR3hsTG1OdmJURWZNQjBHQTFVRUF4TVdkR3h6ClkyRXViM0puTVM1bGVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXoKTURCYU1IWXhDekFKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSApFdzFUWVc0Z1JuSmhibU5wYzJOdk1Sa3dGd1lEVlFRS0V4QnZjbWN4TG1WNFlXMXdiR1V1WTI5dE1SOHdIUVlEClZRUURFeFowYkhOallTNXZjbWN4TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJS29aSXpqMEQKQVFjRFFnQUVmaVhrL1JMRGhJWjRTeFM3NU5hYktkRSszUTBkWVJlU3kySklqQ0YwUk5IRGE5K3Zxa0ZKTHluMQpzVmlzdWgwR0RmQlNjR1NEL056UFZRTVVJZEtqazZOdE1Hc3dEZ1lEVlIwUEFRSC9CQVFEQWdHbU1CMEdBMVVkCkpRUVdNQlFHQ0NzR0FRVUZCd01DQmdnckJnRUZCUWNEQVRBUEJnTlZIUk1CQWY4RUJUQURBUUgvTUNrR0ExVWQKRGdRaUJDQlhwWHptSXE5L05KNUsvT1l6SnV1YVZ4cXlqdFRUWXpXbERzeGVKRXY5Y3pBS0JnZ3Foa2pPUFFRRApBZ05JQURCRkFpQXFYUSs4cHVSZzRqTFUwVmZid2p4clRjMU9xbmEwdHUzYnBtSTRHQVlyUkFJaEFLQ2RiVUI2ClJrVVV4UzVNakpNbmV5bWtLQ3oxNmI0KzZKMzN1ekVSVGJHeQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" ] }, \"type\": 0 }, \"version\": \"0\" } }, \"version\": \"0\" }, \"Org2\": { \"groups\": {}, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Endorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"PEER\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 }, { \"signed_by\": 2 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" }, { \"principal\": { \"msp_identifier\": \"Org2MSP\", \"role\": \"CLIENT\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 } ] } }, \"version\": 0 } }, \"version\": \"0\" } }, \"values\": { \"AnchorPeers\": { \"mod_policy\": \"Admins\", \"value\": { \"anchor_peers\": [ { \"host\": \"127.0.0.1\", \"port\": 29025 } ] }, \"version\": \"0\" }, \"MSP\": { \"mod_policy\": \"Admins\", \"value\": { \"config\": { \"admins\": [], \"crypto_config\": { \"identity_identifier_hash_function\": \"SHA256\", \"signature_hash_family\": \"SHA2\" }, \"fabric_node_ous\": { \"admin_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"admin\" }, \"client_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"client\" }, \"enable\": true, \"orderer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"orderer\" }, \"peer_ou_identifier\": { \"certificate\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\", \"organizational_unit_identifier\": \"peer\" } }, \"intermediate_certs\": [], \"name\": \"Org2MSP\", \"organizational_unit_identifiers\": [], \"revocation_list\": [], \"root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNVVENDQWZlZ0F3SUJBZ0lRWlBOeit0Ym5xT24ydTUrYkNEUjYzREFLQmdncWhrak9QUVFEQWpCek1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWNNQm9HQTFVRUF4TVRZMkV1CmIzSm5NaTVsZUdGdGNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmEKTUhNeEN6QUpCZ05WQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVApZVzRnUm5KaGJtTnBjMk52TVJrd0Z3WURWUVFLRXhCdmNtY3lMbVY0WVcxd2JHVXVZMjl0TVJ3d0dnWURWUVFECkV4TmpZUzV2Y21jeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDQVFZSUtvWkl6ajBEQVFjRFFnQUUKQkpSYXRXSDcxdVJySmEyQWMwT1BuYVl3U3U0bDZKOFdEMjh4K0xyYmtaaHZvWDJoSmJXSkhTQWpodCtLMFNkMwpTYzVLWGUrOHdYTmpQcXdTVWltWlZLTnRNR3N3RGdZRFZSMFBBUUgvQkFRREFnR21NQjBHQTFVZEpRUVdNQlFHCkNDc0dBUVVGQndNQ0JnZ3JCZ0VGQlFjREFUQVBCZ05WSFJNQkFmOEVCVEFEQVFIL01Da0dBMVVkRGdRaUJDQXcKcWtiZEpEdWhQKzZzL2JqcEttay9IZU9kbDJiZlpQRnZaMXFXTitvUlZUQUtCZ2dxaGtqT1BRUURBZ05JQURCRgpBaUVBd1JxL2ZMYU8rUU9wV1p5cGNWSm9jYUxpWnlPQXllS01OOGdmQXVubFBkZ0NJQzVVYW9uZmVkUTYvL3F0ClZsZkRaU0JqbWlYSXJhSVBiWndkd2NLWWRmUTMKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\" ], \"signing_identity\": null, \"tls_intermediate_certs\": [], \"tls_root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNWekNDQWY2Z0F3SUJBZ0lSQU8rVnlBd2YwUWx5MTdaR2xkWk9kU2d3Q2dZSUtvWkl6ajBFQXdJd2RqRUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhHVEFYQmdOVkJBb1RFRzl5WnpJdVpYaGhiWEJzWlM1amIyMHhIekFkQmdOVkJBTVRGblJzCmMyTmhMbTl5WnpJdVpYaGhiWEJzWlM1amIyMHdIaGNOTWpNeE1qSTNNVFkwTXpBd1doY05Nek14TWpJME1UWTAKTXpBd1dqQjJNUXN3Q1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRQpCeE1OVTJGdUlFWnlZVzVqYVhOamJ6RVpNQmNHQTFVRUNoTVFiM0puTWk1bGVHRnRjR3hsTG1OdmJURWZNQjBHCkExVUVBeE1XZEd4elkyRXViM0puTWk1bGVHRnRjR3hsTG1OdmJUQlpNQk1HQnlxR1NNNDlBZ0VHQ0NxR1NNNDkKQXdFSEEwSUFCTi96a2NwRHNwNmM5eTlTOGsxWityeXE4aE5vTjZObjFpOFFtcURMQjJXR2ptWThWam1nSythKwpKWmRqclA2dG1hWlVjYldjcWUxOTIyQm9ZcWhWcGdpamJUQnJNQTRHQTFVZER3RUIvd1FFQXdJQnBqQWRCZ05WCkhTVUVGakFVQmdnckJnRUZCUWNEQWdZSUt3WUJCUVVIQXdFd0R3WURWUjBUQVFIL0JBVXdBd0VCL3pBcEJnTlYKSFE0RUlnUWd1WEI4dDBvTzRnMDdTSHhRMk0rTi9ZZXhTdFJEMFUwSmZHaE0rQmZFdUgwd0NnWUlLb1pJemowRQpBd0lEUndBd1JBSWdDMkVVZW14dDBoaWdQdWZtN0VQVjFWc3U0NU5JSlNRckhDeC9pcmlTK2FFQ0lFbEwvNTRjClhzVnhobVhhVXRmckdrZm5sK01JbllURmpjdjJ3STJLZUdpRAotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" ] }, \"type\": 0 }, \"version\": \"0\" } }, \"version\": \"0\" } }, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Admins\" } }, \"version\": \"0\" }, \"Endorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Endorsement\" } }, \"version\": \"0\" }, \"LifecycleEndorsement\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Endorsement\" } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Readers\" } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Writers\" } }, \"version\": \"0\" } }, \"values\": { \"Capabilities\": { \"mod_policy\": \"Admins\", \"value\": { \"capabilities\": { \"V2_0\": {} } }, \"version\": \"0\" } }, \"version\": \"0\" }, \"Orderer\": { \"groups\": { \"OrdererOrg\": { \"groups\": {}, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"OrdererMSP\", \"role\": \"ADMIN\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"OrdererMSP\", \"role\": \"MEMBER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"msp_identifier\": \"OrdererMSP\", \"role\": \"MEMBER\" }, \"principal_classification\": \"ROLE\" } ], \"rule\": { \"n_out_of\": { \"n\": 1, \"rules\": [ { \"signed_by\": 0 } ] } }, \"version\": 0 } }, \"version\": \"0\" } }, \"values\": { \"Endpoints\": { \"mod_policy\": \"Admins\", \"value\": { \"addresses\": [ \"127.0.0.1:29000\", \"127.0.0.1:29005\", \"127.0.0.1:29010\", \"127.0.0.1:29015\" ] }, \"version\": \"0\" }, \"MSP\": { \"mod_policy\": \"Admins\", \"value\": { \"config\": { \"admins\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNDekNDQWJHZ0F3SUJBZ0lSQUpaUGFLTk9TWG01T2g0bjJLWXNjWjh3Q2dZSUtvWkl6ajBFQXdJd2FURUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJjd0ZRWURWUVFERXc1allTNWxlR0Z0CmNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmFNRll4Q3pBSkJnTlYKQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcApjMk52TVJvd0dBWURWUVFEREJGQlpHMXBia0JsZUdGdGNHeGxMbU52YlRCWk1CTUdCeXFHU000OUFnRUdDQ3FHClNNNDlBd0VIQTBJQUJGWnFPdzJEcWpjaEFPOHpkM0YwNFhBR01xQ0FEeHBRdWZBeFErcGE0Smt3VDVya2VZK2YKNWxwZEZKOWJqWjZEWmEzdlRZNk1PNzhqamI0K2ZEZExrL1dqVFRCTE1BNEdBMVVkRHdFQi93UUVBd0lIZ0RBTQpCZ05WSFJNQkFmOEVBakFBTUNzR0ExVWRJd1FrTUNLQUlLT0V4RURoSmRJLy9rQU9oYnQrcFdrcDQ5VUJuWlRBCmdxSHFESjdqUEdaL01Bb0dDQ3FHU000OUJBTUNBMGdBTUVVQ0lRQ1BxVnRQWndEdHY1TkoxelIvOExzNjJLMjEKSHMxRnhrS3NDTDhqWmJSaUVnSWdPc3JvQ3huNWhXTDVGTlI2Y1NpWEc5bHVYYVE4QkJFeXM5bVJFSk0veGRZPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" ], \"crypto_config\": { \"identity_identifier_hash_function\": \"SHA256\", \"signature_hash_family\": \"SHA2\" }, \"fabric_node_ous\": null, \"intermediate_certs\": [], \"name\": \"OrdererMSP\", \"organizational_unit_identifiers\": [], \"revocation_list\": [], \"root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNQakNDQWVPZ0F3SUJBZ0lRWEpmMjhlcGZiQ0ZUZzlzZXZyL2JRVEFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dhVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4RkRBU0JnTlZCQW9UQzJWNFlXMXdiR1V1WTI5dE1SY3dGUVlEVlFRREV3NWpZUzVsZUdGdGNHeGxMbU52CmJUQlpNQk1HQnlxR1NNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJNSS90NjdhdUlPT1lrOCtyN3p2UkhiRDQ4S0sKWGNtYUdPVmQwbGpyMlY0Q21uQjdCNk4wWTd1d001N0U0QlBMVVFuRmFPZkhvWG1kVThvNGZ1b21GZGFqYlRCcgpNQTRHQTFVZER3RUIvd1FFQXdJQnBqQWRCZ05WSFNVRUZqQVVCZ2dyQmdFRkJRY0RBZ1lJS3dZQkJRVUhBd0V3CkR3WURWUjBUQVFIL0JBVXdBd0VCL3pBcEJnTlZIUTRFSWdRZ280VEVRT0VsMGovK1FBNkZ1MzZsYVNuajFRR2QKbE1DQ29lb01udU04Wm44d0NnWUlLb1pJemowRUF3SURTUUF3UmdJaEFKSkJCWUI5KzVKWmZvdGR1TkxXcTBPWAo4WTlEYml0QUQxa0ZXL2xVaHUxbEFpRUFyQWc3WXpmVGdPL1o2Nlg3d0UwVmJ0cWQ1UXFzRzZQZFBFdlViY21ICmZCaz0KLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=\" ], \"signing_identity\": null, \"tls_intermediate_certs\": [], \"tls_root_certs\": [ \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNRakNDQWVtZ0F3SUJBZ0lRTnVQWjlaREJTcloxclMxd3dDS00vREFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93YkRFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEZEQVNCZ05WQkFvVEMyVjRZVzF3YkdVdVkyOXRNUm93R0FZRFZRUURFeEYwYkhOallTNWxlR0Z0CmNHeGxMbU52YlRCWk1CTUdCeXFHU000OUFnRUdDQ3FHU000OUF3RUhBMElBQk5nTEpBQ0NPb3MwdGo1Ri96UUoKak81a2IrWm5oKzVyK0NrL01SOXU0VFVnVm03MHdYZXJyQlFra0s2MzdFbXFlMnFvQWY1V1JqeFVULzdEcVB0ZgpnWG1qYlRCck1BNEdBMVVkRHdFQi93UUVBd0lCcGpBZEJnTlZIU1VFRmpBVUJnZ3JCZ0VGQlFjREFnWUlLd1lCCkJRVUhBd0V3RHdZRFZSMFRBUUgvQkFVd0F3RUIvekFwQmdOVkhRNEVJZ1FnallLSndVWjB5NnhXWFZQVnFVR1EKdUE5eWxuLzZHVXh5WDRIeUdVTURtMWt3Q2dZSUtvWkl6ajBFQXdJRFJ3QXdSQUlnUzFKdFdUenFNMG8zMHVqTQpYeTJXOTVnTzcvdUNnMVpVWG9ZWlo5OTdUemdDSUg0dU8xOWZXL2k2UTNSd09tdk1uRm5DVWo0dFNWVFpzSXhICk5tNzdHSGJPCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\" ] }, \"type\": 0 }, \"version\": \"0\" } }, \"version\": \"0\" } }, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Admins\" } }, \"version\": \"0\" }, \"BlockValidation\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 1, \"value\": { \"identities\": [ { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRQ3lkRXdicW43Yy9QbU1kMFV1Sm1UakFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl4TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUV3b1U0ejRVMEZTUTFQaUI1Qm5XUDVEZ1BOQVlEc2QrUVhiQ3VPTndrM3RWNGkvaGgKZnh5aVFaS0RzUEUzUU5HV0JuaFFrSVpuZVF0QTQwVWNDTEdmMjZOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTjZ6ZCtJRXVEcUViMVJiakh6b2tKaUMKdUZRYm1wa2Y1aUhkSnc5Zm5kMDNBaUFkNEVZNmZvVTAzSG9vN2N1TjVsN2tubnZVVnFtZWlUZEVJVTE2N0g1cQpXZz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" }, { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEakNDQWJTZ0F3SUJBZ0lSQU4rZzVTNkZFMk1ldlNaZ1ZCT3Aybmd3Q2dZSUtvWkl6ajBFQXdJd2FURUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJjd0ZRWURWUVFERXc1allTNWxlR0Z0CmNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmFNRmt4Q3pBSkJnTlYKQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcApjMk52TVIwd0d3WURWUVFERXhSdmNtUmxjbVZ5TWk1bGVHRnRjR3hsTG1OdmJUQlpNQk1HQnlxR1NNNDlBZ0VHCkNDcUdTTTQ5QXdFSEEwSUFCQzFaNTlSbmNYdVI1OG9DVzErV1RWK1hsOGpxY2NCbzVkbDB5L0h6VDZrd3JDQzcKTytmK2x5bEFvZTgxTGdqdVIvNEtMcTBLQ0pVZnhnZkpGMDlXSGV5alRUQkxNQTRHQTFVZER3RUIvd1FFQXdJSApnREFNQmdOVkhSTUJBZjhFQWpBQU1Dc0dBMVVkSXdRa01DS0FJS09FeEVEaEpkSS8va0FPaGJ0K3BXa3A0OVVCCm5aVEFncUhxREo3alBHWi9NQW9HQ0NxR1NNNDlCQU1DQTBnQU1FVUNJUUNiWWxjaU9WaGpueUorQmhoYXZNQlcKKzlzUGRGYnh4MFpEWjdtVlNHT1Z6QUlnRzVkS2wzemV0bS9yRXBpS2ZjOXNTVTZ1OXgrMWJTTGtiR1RCQnpCYgpWYkU9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" }, { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNDekNDQWJPZ0F3SUJBZ0lRZTdObHVkYW9oeHhwbmdhM3padGlyekFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl6TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVlTHJpdnNOT0VoT2xTTyt3eFNGemlzU3R2VGRZa2taTDRnaG1VN0VmVVo0K0hsL3AKWU0wZ28xZHlDOFRyYW9sNDh6cnhidXAwSXk3TzNuSmpiOWdaQnFOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEUmdBd1F3SWZKREJwbURDWC9WSEFCNmd1andFT3FxdUkKZEtaRGxzWklaWHAwTkVLb3dBSWdhVWJMNlZ4dkx0Q2dYNjdCMDBFa1ZzOXNjZDMxTHFxYlZTcHBOZnR2bVd3PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" }, { \"principal\": { \"id_bytes\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRRGJvNFB5VzlObnIwOSsvMUcyeHBFREFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEkwTG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVqaWlJdkU4cG5qZlNZeGZCdURDbExqOUdZNXRYb2k0MmpBcVdjRE5UV2JYYTI4NVYKb0VXd2FDenpKeXF1WUE5QzZXM2pRQTV1U1dtWjNBWVlKUFZGV3FOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTThWQWFCTHF2TU14bEI0SzlsSE44RjYKNFhwWlU5ZmxPS3pLUHowdFFEQ1pBaUFYSzJoM1FuaFAyb2VGT21HYzYreE1oVHhOVDFYemxxZWF2ZWlWamRpZgpqdz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"mspid\": \"OrdererMSP\" }, \"principal_classification\": \"IDENTITY\" } ], \"rule\": { \"n_out_of\": { \"n\": 3, \"rules\": [ { \"signed_by\": 0 }, { \"signed_by\": 1 }, { \"signed_by\": 2 }, { \"signed_by\": 3 } ] } }, \"version\": 0 } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Readers\" } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Writers\" } }, \"version\": \"0\" } }, \"values\": { \"BatchSize\": { \"mod_policy\": \"Admins\", \"value\": { \"absolute_max_bytes\": 1000000, \"max_message_count\": 1, \"preferred_max_bytes\": 524288 }, \"version\": \"1\" }, \"BatchTimeout\": { \"mod_policy\": \"Admins\", \"value\": { \"timeout\": \"1s\" }, \"version\": \"0\" }, \"Capabilities\": { \"mod_policy\": \"Admins\", \"value\": { \"capabilities\": { \"V2_0\": {} } }, \"version\": \"0\" }, \"ChannelRestrictions\": { \"mod_policy\": \"Admins\", \"value\": null, \"version\": \"0\" }, \"ConsensusType\": { \"mod_policy\": \"Admins\", \"value\": { \"metadata\": { \"collect_timeout\": \"1s\", \"decisions_per_leader\": \"0\", \"incoming_message_buffer_size\": \"200\", \"leader_heartbeat_count\": \"10\", \"leader_heartbeat_timeout\": \"1m0s\", \"leader_rotation\": \"ROTATION_UNSPECIFIED\", \"request_auto_remove_timeout\": \"3m0s\", \"request_batch_max_bytes\": \"10485760\", \"request_batch_max_count\": \"100\", \"request_batch_max_interval\": \"50ms\", \"request_complain_timeout\": \"20s\", \"request_forward_timeout\": \"2s\", \"request_max_bytes\": \"0\", \"request_pool_size\": \"400\", \"speed_up_view_change\": false, \"sync_on_start\": false, \"view_change_resend_interval\": \"5s\", \"view_change_timeout\": \"20s\" }, \"state\": \"STATE_NORMAL\", \"type\": \"BFT\" }, \"version\": \"0\" }, \"Orderers\": { \"mod_policy\": \"Admins\", \"value\": { \"consenter_mapping\": [ { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlhZ0F3SUJBZ0lSQU5rMFcrNTBwdUJkSmFFK0VmK284c3d3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQktiSXFLZTVZbG5vY3lCV3lJczZpaHJpQUY3UllqSjZXN2lDS094b3YyV1EKbHNHazA1R2c5N21lcnBsOEZpSEJGSjJ3cWs5cDhXSG9HbWVkUFVoNlQrU2pnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJeExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlNWUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwY0FNRVFDCklCandvaitiV1ZkeTZQMUlBMGw3a2FpSEYwOGhyQTRCRTY0SktLZzZyblhQQWlCOUg1NEVwZTI0L2JTWkwvNC8KVjdnY2pIQ3lhcXVwUEp6R3M2VFdGSnhYSUE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"host\": \"127.0.0.1\", \"id\": 1, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRQ3lkRXdicW43Yy9QbU1kMFV1Sm1UakFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl4TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUV3b1U0ejRVMEZTUTFQaUI1Qm5XUDVEZ1BOQVlEc2QrUVhiQ3VPTndrM3RWNGkvaGgKZnh5aVFaS0RzUEUzUU5HV0JuaFFrSVpuZVF0QTQwVWNDTEdmMjZOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTjZ6ZCtJRXVEcUViMVJiakh6b2tKaUMKdUZRYm1wa2Y1aUhkSnc5Zm5kMDNBaUFkNEVZNmZvVTAzSG9vN2N1TjVsN2tubnZVVnFtZWlUZEVJVTE2N0g1cQpXZz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"msp_id\": \"OrdererMSP\", \"port\": 29003, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlhZ0F3SUJBZ0lSQU5rMFcrNTBwdUJkSmFFK0VmK284c3d3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU1TNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQktiSXFLZTVZbG5vY3lCV3lJczZpaHJpQUY3UllqSjZXN2lDS094b3YyV1EKbHNHazA1R2c5N21lcnBsOEZpSEJGSjJ3cWs5cDhXSG9HbWVkUFVoNlQrU2pnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJeExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlNWUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwY0FNRVFDCklCandvaitiV1ZkeTZQMUlBMGw3a2FpSEYwOGhyQTRCRTY0SktLZzZyblhQQWlCOUg1NEVwZTI0L2JTWkwvNC8KVjdnY2pIQ3lhcXVwUEp6R3M2VFdGSnhYSUE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" }, { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlXZ0F3SUJBZ0lRZUlVcGZPd3RGR0tLQ0Q3MFRKN1NOekFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFRjNkNFhzcG5FVk94Y2lnekJmMEg1TWFxY1JsMHBLYWV0alZXdHpIZzRGZGYKYkcyQXFTdE1VcG0xYmNkb3VEZFNESUd0ZUFIckJxcFAwNWZXcXhKK0NhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pJdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXlnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFNBQXdSUUloCkFNK2szcE9xa2YrNUlvcGplSW5FaDY0djRwd0t5UWc4cG1FU3E2cmNGV1BJQWlBaXUrMEZnSTNlcDNRTDg1emwKdmsySjdxRmhsUHdyL28rYzQxUzQxaVpuSkE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"host\": \"127.0.0.1\", \"id\": 2, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEakNDQWJTZ0F3SUJBZ0lSQU4rZzVTNkZFMk1ldlNaZ1ZCT3Aybmd3Q2dZSUtvWkl6ajBFQXdJd2FURUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJjd0ZRWURWUVFERXc1allTNWxlR0Z0CmNHeGxMbU52YlRBZUZ3MHlNekV5TWpjeE5qUXpNREJhRncwek16RXlNalF4TmpRek1EQmFNRmt4Q3pBSkJnTlYKQkFZVEFsVlRNUk13RVFZRFZRUUlFd3BEWVd4cFptOXlibWxoTVJZd0ZBWURWUVFIRXcxVFlXNGdSbkpoYm1OcApjMk52TVIwd0d3WURWUVFERXhSdmNtUmxjbVZ5TWk1bGVHRnRjR3hsTG1OdmJUQlpNQk1HQnlxR1NNNDlBZ0VHCkNDcUdTTTQ5QXdFSEEwSUFCQzFaNTlSbmNYdVI1OG9DVzErV1RWK1hsOGpxY2NCbzVkbDB5L0h6VDZrd3JDQzcKTytmK2x5bEFvZTgxTGdqdVIvNEtMcTBLQ0pVZnhnZkpGMDlXSGV5alRUQkxNQTRHQTFVZER3RUIvd1FFQXdJSApnREFNQmdOVkhSTUJBZjhFQWpBQU1Dc0dBMVVkSXdRa01DS0FJS09FeEVEaEpkSS8va0FPaGJ0K3BXa3A0OVVCCm5aVEFncUhxREo3alBHWi9NQW9HQ0NxR1NNNDlCQU1DQTBnQU1FVUNJUUNiWWxjaU9WaGpueUorQmhoYXZNQlcKKzlzUGRGYnh4MFpEWjdtVlNHT1Z6QUlnRzVkS2wzemV0bS9yRXBpS2ZjOXNTVTZ1OXgrMWJTTGtiR1RCQnpCYgpWYkU9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"msp_id\": \"OrdererMSP\", \"port\": 29008, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmekNDQWlXZ0F3SUJBZ0lRZUlVcGZPd3RGR0tLQ0Q3MFRKN1NOekFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJeUxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFRjNkNFhzcG5FVk94Y2lnekJmMEg1TWFxY1JsMHBLYWV0alZXdHpIZzRGZGYKYkcyQXFTdE1VcG0xYmNkb3VEZFNESUd0ZUFIckJxcFAwNWZXcXhKK0NhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pJdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXlnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFNBQXdSUUloCkFNK2szcE9xa2YrNUlvcGplSW5FaDY0djRwd0t5UWc4cG1FU3E2cmNGV1BJQWlBaXUrMEZnSTNlcDNRTDg1emwKdmsySjdxRmhsUHdyL28rYzQxUzQxaVpuSkE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" }, { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmakNDQWlXZ0F3SUJBZ0lRVGRJa1Q1TU5YVW8yNXI4NUJNbHVQakFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJekxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFeEowOTE1L2N1Rk5RczdLeUhCV3JyWTVHb1hvWkxMc1VjRHJMTFBhd21odmwKemllbG9nYVE1Tll1WmFYVk5IRm5hcFAvTy94K09Ncmlvejh2ZDNhWkhhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pNdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXpnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFJ3QXdSQUlnCmVNTnNVS29lMmdXQnpVdzRIeWpkN3FqeUNlUmwxaUtwejQxcmtGSnBNRWNDSUhtdXp1Yng1ZUljdGd5ZHNaUHMKR0xvOERlSzNPMEVWdHJESkNrQ2YzL2RrCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"host\": \"127.0.0.1\", \"id\": 3, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNDekNDQWJPZ0F3SUJBZ0lRZTdObHVkYW9oeHhwbmdhM3padGlyekFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEl6TG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVlTHJpdnNOT0VoT2xTTyt3eFNGemlzU3R2VGRZa2taTDRnaG1VN0VmVVo0K0hsL3AKWU0wZ28xZHlDOFRyYW9sNDh6cnhidXAwSXk3TzNuSmpiOWdaQnFOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEUmdBd1F3SWZKREJwbURDWC9WSEFCNmd1andFT3FxdUkKZEtaRGxzWklaWHAwTkVLb3dBSWdhVWJMNlZ4dkx0Q2dYNjdCMDBFa1ZzOXNjZDMxTHFxYlZTcHBOZnR2bVd3PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"msp_id\": \"OrdererMSP\", \"port\": 29013, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNmakNDQWlXZ0F3SUJBZ0lRVGRJa1Q1TU5YVW8yNXI4NUJNbHVQakFLQmdncWhrak9QUVFEQWpCc01Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4R2pBWUJnTlZCQU1URVhSc2MyTmhMbVY0CllXMXdiR1V1WTI5dE1CNFhEVEl6TVRJeU56RTJORE13TUZvWERUTXpNVEl5TkRFMk5ETXdNRm93V1RFTE1Ba0cKQTFVRUJoTUNWVk14RXpBUkJnTlZCQWdUQ2tOaGJHbG1iM0p1YVdFeEZqQVVCZ05WQkFjVERWTmhiaUJHY21GdQpZMmx6WTI4eEhUQWJCZ05WQkFNVEZHOXlaR1Z5WlhJekxtVjRZVzF3YkdVdVkyOXRNRmt3RXdZSEtvWkl6ajBDCkFRWUlLb1pJemowREFRY0RRZ0FFeEowOTE1L2N1Rk5RczdLeUhCV3JyWTVHb1hvWkxMc1VjRHJMTFBhd21odmwKemllbG9nYVE1Tll1WmFYVk5IRm5hcFAvTy94K09Ncmlvejh2ZDNhWkhhT0J1ekNCdURBT0JnTlZIUThCQWY4RQpCQU1DQmFBd0hRWURWUjBsQkJZd0ZBWUlLd1lCQlFVSEF3RUdDQ3NHQVFVRkJ3TUNNQXdHQTFVZEV3RUIvd1FDCk1BQXdLd1lEVlIwakJDUXdJb0FnallLSndVWjB5NnhXWFZQVnFVR1F1QTl5bG4vNkdVeHlYNEh5R1VNRG0xa3cKVEFZRFZSMFJCRVV3UTRJVWIzSmtaWEpsY2pNdVpYaGhiWEJzWlM1amIyMkNDRzl5WkdWeVpYSXpnZ2xzYjJOaApiR2h2YzNTSEJIOEFBQUdIRUFBQUFBQUFBQUFBQUFBQUFBQUFBQUV3Q2dZSUtvWkl6ajBFQXdJRFJ3QXdSQUlnCmVNTnNVS29lMmdXQnpVdzRIeWpkN3FqeUNlUmwxaUtwejQxcmtGSnBNRWNDSUhtdXp1Yng1ZUljdGd5ZHNaUHMKR0xvOERlSzNPMEVWdHJESkNrQ2YzL2RrCi0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\" }, { \"client_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNnRENDQWlhZ0F3SUJBZ0lSQU8vRjJYbjBPMi9CaDVxYlRwQWdkVTB3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU5DNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQkpKeDdCZXlpK3Rtbk9zM3pSMTVlWURYa0M0WEU0MHJYeWg4Ylpsa1pDeXIKbG4wcVBZZDMybzZ1WlRDVThyRFZsejhyaTBzT2RLUVNvNTcvUkRneVZ5ZWpnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJMExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlOSUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwZ0FNRVVDCklRREJ0ZjY1eGdUYTJuREFVZjZiN1JCWlVKZ29GYlNzTkVINFFvaWJydDlWNHdJZ0g0cWJRQ2tWcEo0UkFKZm8KZUNMNEZ0Qk5ySE9CV0xqamt6NGxIYmdmVWYwPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\", \"host\": \"127.0.0.1\", \"id\": 4, \"identity\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNEVENDQWJPZ0F3SUJBZ0lRRGJvNFB5VzlObnIwOSsvMUcyeHBFREFLQmdncWhrak9QUVFEQWpCcE1Rc3cKQ1FZRFZRUUdFd0pWVXpFVE1CRUdBMVVFQ0JNS1EyRnNhV1p2Y201cFlURVdNQlFHQTFVRUJ4TU5VMkZ1SUVaeQpZVzVqYVhOamJ6RVVNQklHQTFVRUNoTUxaWGhoYlhCc1pTNWpiMjB4RnpBVkJnTlZCQU1URG1OaExtVjRZVzF3CmJHVXVZMjl0TUI0WERUSXpNVEl5TnpFMk5ETXdNRm9YRFRNek1USXlOREUyTkRNd01Gb3dXVEVMTUFrR0ExVUUKQmhNQ1ZWTXhFekFSQmdOVkJBZ1RDa05oYkdsbWIzSnVhV0V4RmpBVUJnTlZCQWNURFZOaGJpQkdjbUZ1WTJsegpZMjh4SFRBYkJnTlZCQU1URkc5eVpHVnlaWEkwTG1WNFlXMXdiR1V1WTI5dE1Ga3dFd1lIS29aSXpqMENBUVlJCktvWkl6ajBEQVFjRFFnQUVqaWlJdkU4cG5qZlNZeGZCdURDbExqOUdZNXRYb2k0MmpBcVdjRE5UV2JYYTI4NVYKb0VXd2FDenpKeXF1WUE5QzZXM2pRQTV1U1dtWjNBWVlKUFZGV3FOTk1Fc3dEZ1lEVlIwUEFRSC9CQVFEQWdlQQpNQXdHQTFVZEV3RUIvd1FDTUFBd0t3WURWUjBqQkNRd0lvQWdvNFRFUU9FbDBqLytRQTZGdTM2bGFTbmoxUUdkCmxNQ0NvZW9NbnVNOFpuOHdDZ1lJS29aSXpqMEVBd0lEU0FBd1JRSWhBTThWQWFCTHF2TU14bEI0SzlsSE44RjYKNFhwWlU5ZmxPS3pLUHowdFFEQ1pBaUFYSzJoM1FuaFAyb2VGT21HYzYreE1oVHhOVDFYemxxZWF2ZWlWamRpZgpqdz09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K\", \"msp_id\": \"OrdererMSP\", \"port\": 29018, \"server_tls_cert\": \"LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNnRENDQWlhZ0F3SUJBZ0lSQU8vRjJYbjBPMi9CaDVxYlRwQWdkVTB3Q2dZSUtvWkl6ajBFQXdJd2JERUwKTUFrR0ExVUVCaE1DVlZNeEV6QVJCZ05WQkFnVENrTmhiR2xtYjNKdWFXRXhGakFVQmdOVkJBY1REVk5oYmlCRwpjbUZ1WTJselkyOHhGREFTQmdOVkJBb1RDMlY0WVcxd2JHVXVZMjl0TVJvd0dBWURWUVFERXhGMGJITmpZUzVsCmVHRnRjR3hsTG1OdmJUQWVGdzB5TXpFeU1qY3hOalF6TURCYUZ3MHpNekV5TWpReE5qUXpNREJhTUZreEN6QUoKQmdOVkJBWVRBbFZUTVJNd0VRWURWUVFJRXdwRFlXeHBabTl5Ym1saE1SWXdGQVlEVlFRSEV3MVRZVzRnUm5KaApibU5wYzJOdk1SMHdHd1lEVlFRREV4UnZjbVJsY21WeU5DNWxlR0Z0Y0d4bExtTnZiVEJaTUJNR0J5cUdTTTQ5CkFnRUdDQ3FHU000OUF3RUhBMElBQkpKeDdCZXlpK3Rtbk9zM3pSMTVlWURYa0M0WEU0MHJYeWg4Ylpsa1pDeXIKbG4wcVBZZDMybzZ1WlRDVThyRFZsejhyaTBzT2RLUVNvNTcvUkRneVZ5ZWpnYnN3Z2Jnd0RnWURWUjBQQVFILwpCQVFEQWdXZ01CMEdBMVVkSlFRV01CUUdDQ3NHQVFVRkJ3TUJCZ2dyQmdFRkJRY0RBakFNQmdOVkhSTUJBZjhFCkFqQUFNQ3NHQTFVZEl3UWtNQ0tBSUkyQ2ljRkdkTXVzVmwxVDFhbEJrTGdQY3BaLytobE1jbCtCOGhsREE1dFoKTUV3R0ExVWRFUVJGTUVPQ0ZHOXlaR1Z5WlhJMExtVjRZVzF3YkdVdVkyOXRnZ2h2Y21SbGNtVnlOSUlKYkc5agpZV3hvYjNOMGh3Ui9BQUFCaHhBQUFBQUFBQUFBQUFBQUFBQUFBQUFCTUFvR0NDcUdTTTQ5QkFNQ0EwZ0FNRVVDCklRREJ0ZjY1eGdUYTJuREFVZjZiN1JCWlVKZ29GYlNzTkVINFFvaWJydDlWNHdJZ0g0cWJRQ2tWcEo0UkFKZm8KZUNMNEZ0Qk5ySE9CV0xqamt6NGxIYmdmVWYwPQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==\" } ] }, \"version\": \"0\" } }, \"version\": \"0\" } }, \"mod_policy\": \"Admins\", \"policies\": { \"Admins\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"MAJORITY\", \"sub_policy\": \"Admins\" } }, \"version\": \"0\" }, \"Readers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Readers\" } }, \"version\": \"0\" }, \"Writers\": { \"mod_policy\": \"Admins\", \"policy\": { \"type\": 3, \"value\": { \"rule\": \"ANY\", \"sub_policy\": \"Writers\" } }, \"version\": \"0\" } }, \"values\": { \"BlockDataHashingStructure\": { \"mod_policy\": \"Admins\", \"value\": { \"width\": 4294967295 }, \"version\": \"0\" }, \"Capabilities\": { \"mod_policy\": \"Admins\", \"value\": { \"capabilities\": { \"V3_0\": {} } }, \"version\": \"0\" }, \"HashingAlgorithm\": { \"mod_policy\": \"Admins\", \"value\": { \"name\": \"SHA256\" }, \"version\": \"0\" }, \"OrdererAddresses\": { \"mod_policy\": \"/Channel/Orderer/Admins\", \"value\": { \"addresses\": [ \"127.0.0.1:29000\", \"127.0.0.1:29005\", \"127.0.0.1:29010\", \"127.0.0.1:29015\" ] }, \"version\": \"0\" } }, \"version\": \"0\" }, \"sequence\": \"1\" } "

	b1 := bytes.NewBufferString(s)
	b2 := bytes.NewBufferString(s)

	c1 := &cb.Config{}
	c2 := &cb.Config{}
	err := protolator.DeepUnmarshalJSON(b1, c1)
	assert.NoError(t, err)
	err = protolator.DeepUnmarshalJSON(b2, c2)
	assert.NoError(t, err)

	// SUCCESS
	assert.True(t, proto.Equal(c1, c2))

	m1, err := proto.Marshal(c1)
	assert.NoError(t, err)
	m2, err := proto.Marshal(c2)
	assert.NoError(t, err)

	txID1 := sha256.Sum256(m1)
	txID2 := sha256.Sum256(m2)
	TxIDStr1 := hex.EncodeToString(txID1[:])
	TxIDStr2 := hex.EncodeToString(txID2[:])

	// FAIL !!!
	assert.Equal(t, m1, m2)

	// FAIL !!!
	assert.Equal(t, TxIDStr1, TxIDStr2)
}
```

Reply:
That's the reason: map in go

<img width="974" alt="1" src="https://github.com/hyperledger/fabric/assets/7284783/130ebcc3-3869-4e9b-844b-defb982a1c7e">

Reply:
Is this the change you made?

```
diff --git a/orderer/consensus/smartbft/util.go b/orderer/consensus/smartbft/util.go
index e2bf9779c..f3da100ff 100644
--- a/orderer/consensus/smartbft/util.go
+++ b/orderer/consensus/smartbft/util.go
@@ -13,6 +13,7 @@ import (
        "encoding/hex"
        "encoding/pem"
        "fmt"
+       "github.com/hyperledger/fabric/common/util"
        "sort"
        "time"
 
@@ -249,6 +250,7 @@ type request struct {
        sigHdr   *cb.SignatureHeader
        envelope *cb.Envelope
        chHdr    *cb.ChannelHeader
+       data     []byte
 }
 
 // RequestInspector inspects incomming requests and validates serialized identity
@@ -283,6 +285,14 @@ func (ri *RequestInspector) RequestID(rawReq []byte) types.RequestInfo {
        if err != nil {
                return types.RequestInfo{}
        }
+
+       if req.chHdr.Type == int32(cb.HeaderType_CONFIG) {
+               return types.RequestInfo{
+                       ID:       hex.EncodeToString(util.ComputeSHA256(req.data)),
+                       ClientID: "",
+               }
+       }
+
        reqInfo, err := ri.requestIDFromSigHeader(req.sigHdr)
        if err != nil {
                return types.RequestInfo{}
@@ -319,6 +329,7 @@ func (ri *RequestInspector) unwrapReq(req []byte) (*request, error) {
        }
 
        return &request{
+               data:     payload.Data,
                chHdr:    chdr,
                sigHdr:   sigHdr,
                envelope: envelope,

```

Can you tell me in which integration test it fails?

Reply:
Not a fail. I needed to check what RequestInfo is returned when calculating the hash from payload.Data.
I used the smartbft test "send an update transaction to each orderer and deleting failed requests". It failed at the end, then it outputs a log. And in the log I found records "Request %s was submitted" and saw that RequestInfo is all different.

In my case they are the same.

Reply:
In your case, this is the result:

```
    STEP: Begin broadcast 12/28/23 01:53:58.271
    [e][OrdererOrg.orderer1] 2023-12-28 01:53:58.290 MSK 0282 DEBU [orderer.consensus.smartbft.consensus] addRequest -> Request { 6ff781af2b9804de21e80e3f887737bd182940790de21a802578d3a52717d0f0} was submitted channel=testchannel1
    [e][OrdererOrg.orderer3] 2023-12-28 01:53:58.291 MSK 021e DEBU [orderer.consensus.smartbft.consensus] addRequest -> Request { 6e25d304cf3805641a23f565e4b57ce9b686557b146ac5ab35a3454f5009db3e} was submitted channel=testchannel1
    [e][OrdererOrg.orderer4] 2023-12-28 01:53:58.292 MSK 020b DEBU [orderer.consensus.smartbft.consensus] addRequest -> Request { aef2933821684c70a5213cd57824c7f1eafa07b5c0d1fb063960fcd59d2e9764} was submitted channel=testchannel1
    [e][OrdererOrg.orderer2] 2023-12-28 01:53:58.292 MSK 025e DEBU [orderer.consensus.smartbft.consensus] addRequest -> Request { 48ee699b67a0cef7799cc3f64e6536202973cd61e79d3fa7dc4bd260e3941efd} was submitted channel=testchannel1
```

Reply:
Yeah you're right, I see what's going on here: When we pass the config update through the broadcast layer, it creates the config snapshot from the current state and then when it marshals it into bytes, it's non deterministic as you pointed out, since it has protobuf maps.

Using the latest update indeed lets us only capture the context of the intent of the transactor and not the entire update afterwards. 

However, after thinking about it, I think we should not only include the latest update, but also the sequence(`ConfigEnvelope.Config.Sequence`), otherwise we can do a denial of service attack by re-submitting an old config update with an older sequence and this prevent a new config transaction with the same content (`LastUpdate`) from getting into the request pool (since the transaction identifier will be the same). 


Thoughts?

Reply:
I will describe the basic algorithm for changing the channel config.
1. User takes the current channel config
2. Modifies it
3. Using configtxlator or the `upd, err := update.Compute(conf, newConf)` function, the difference of the changes is calculated.
3a. In the change difference with each changed section is the Version field, which is incremented by one.
3b. In essence, the resulting RWSet is like this. When such a change segment hits the orderer, if the new version of the section has already been applied or incremented by more than one, such an update will be rejected.
4. the update (change difference) is signed according to policy
5. An envelope is prepared and sent to the Warrant Officer
6. The current channel config is taken at the orderer and changes are applied to it with those section versions that were generated at the client. 
7. At the orderer the envelope with the difference of changes, if it is accepted, is added to the `LastUpdate` field in the unchanged form.

How do you conduct a denial-of-service attack here?

If you send an old "LastUpdate", when you try to apply it to the current channel configuration, it will most likely fail because of the versions of the changed sections

Reply:
So far it seems to me that such an attack would not cause problems.
But if you come up with a scenario, I'm willing to write a test, and test it.

Reply:
I was thinking that if you take an old config update and re-submit it, it will take the same txID as a new config and will prevent config transactions with the same content from entering the request pool.

But now after looking at the code, I see that's not possible because the broadcast filters will make sure the global sequence is exactly the latest one plus one.